File: Programmers/JAVA/lv1/lv1_콜라츠 추측.java
https://programmers.co.kr/learn/courses/30/lessons/12943?language=java

class Solution {
    public int solution(long num) {
        int answer = 0;
        int count = 0;
        
        while(num != 1){
            count++;
            if(count != 500){
                num = num%2==0 ? num/2 : num*3+1;   
            }
            else {
                count = -1;
                break;
            }
        }
        answer = count;
        
        return answer;
    }
}

File: Programmers/C#/lv1/lv1_콜라츠 추측.cs
https://programmers.co.kr/learn/courses/30/lessons/12943?language=csharp

public class Solution
{
    public int solution(int num)
    {
        int answer = -1;
        
        long n = num;
        int ctr = 0;
        if (n > 1)
        {
            while (ctr < 500)
            {
                if (n % 2 == 0)
                    n = n / 2;
                else
                    n = n * 3 + 1;

                ctr++;
                if (n == 1)
                {
                    answer = ctr;
                    break;
                }
            }
        }
        else
            return 0;

        return answer;
    }
}